File: nodejs_samples/create_ticket.js
// Requires unirest module. To install:
// npm install unirest

var unirest = require('unirest');

var API_KEY = "YOUR_API_KEY";
var FD_ENDPOINT = "YOUR_DOMAIN";

var PATH = "/api/v2/tickets";
var URL =  "https://" + FD_ENDPOINT + ".freshdesk.com"+ PATH;

var fields = {
  'email': 'email@yourdomain.com',
  'subject': 'Ticket subject',
  'description': 'Ticket description.',
  'status': 2,
  'priority': 1
}

var Request = unirest.post(URL);

Request.auth({
  user: API_KEY,
  pass: "X",
  sendImmediately: true
})
.type('json')
.send(fields)
.end(function(response){
  console.log(response.body)
  console.log("Response Status : " + response.status)
  if(response.status == 201){
    console.log("Location Header : "+ response.headers['location'])
  }
  else{
    	console.log("X-Request-Id :" + response.headers['x-request-id']);
  }
  });

File: Python/create_ticket.py
## This script requires "requests": http://docs.python-requests.org/
## To install: pip install requests

import requests
import json

api_key = "YOUR_API_KEY"
domain = "YOUR_DOMAIN"
password = "x"

headers = { 'Content-Type' : 'application/json' }

ticket = {
    'subject' : 'Ticket title',
    'description' : 'Ticket detail',
    'email' : 'example@example.com',
    'priority' : 1,
    'status' : 2,
    'cc_emails' : ['sample_email@domain.com', 'user_email@domain.com']
}

r = requests.post("https://"+ domain +".freshdesk.com/api/v2/tickets", auth = (api_key, password), headers = headers, data = json.dumps(ticket))

if r.status_code == 201:
  print "Ticket created successfully, the response is given below" + r.content
  print "Location Header : " + r.headers['Location']
else:
  print "Failed to create ticket, errors are displayed below,"
  response = json.loads(r.content)
  print response["errors"]

  print "x-request-id : " + r.headers['x-request-id']
  print "Status Code : " + str(r.status_code)